### PR TITLE
KREST-360: Start brokers concurrently in ClusterTestHarness

### PR DESCRIPTION
**Pull request by agg111**
### What
The integration tests in kafka rest use ClusterTestHarness that has set up to bring up the services. This PR concerns modifying the way kafka servers are set up. Currently, the brokers are started sequentially which is a drawback to have topics with replication_factor > 1. This PR works on concurrently starting brokers.

### Testing
Making sure all tests pass.

### References
Jira - https://confluentinc.atlassian.net/browse/KREST-360

### CODE REVIEW

File: kafka-rest/src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java [hunk 1]
```diff
@@ -239,6 +234,41 @@ public void setUp() throws Exception {
     restServer.start();
   }
 
+  private void startBrokersConcurrently(int numBrokers, Duration timeout) throws Exception {
+    List<Future<KafkaServer>> serverFutures = new ArrayList<>(numBrokers);
+    configs = new Vector<>();
```
Comment (rigelbm) [inline kafka-rest/src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java:239]: You can compute all configs sequentially, prior to computing the servers.

```
configs =
    IntStream.range(0, numBrokers)
        .map(brokerId -> KafkaConfig.fromProps(getBrokerProperties(brokerId)))
        .collect(toList());
```

File: kafka-rest/src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java [hunk 2]
```diff
@@ -239,6 +234,41 @@ public void setUp() throws Exception {
...
+    servers = new Vector<>();
```
Comment (rigelbm) [inline kafka-rest/src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java:240]: You can achieve the same as everything below with this one-liner.

```
servers =
    CompletableFutures.allAsList(
        configs.stream()
            .map(
                config ->
                    CompletableFuture.supplyAsync(
                        () -> TestUtils.createServer(
                            config,
                            new MockTime(System.currentTimeMillis(),
                            System.nanoTime())))
            .collect(toList()))
        .join();
```

File: kafka-rest/src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java [hunk 3]
```diff
@@ -239,6 +237,25 @@ public void setUp() throws Exception {
     restServer.start();
   }
 
+  private void startBrokersConcurrently(int numBrokers) {
+    configs =
+        IntStream.range(0, numBrokers)
+            .mapToObj(brokerId -> KafkaConfig.fromProps(getBrokerProperties(brokerId)))
```
Comment (rigelbm) [inline kafka-rest/src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java:243]: Should be `KafkaConfig.fromProps(overrideBrokerProperties(brokerId, getBrokerProperties(brokerId)))`. I think that's why the tests are failing.

Comment (agg111) [reply kafka-rest/src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java]: Added call to overrideBrokerProperties.

**Review by rigelbm — APPROVED** [review]
LGTM

**Pull request merged**